### PR TITLE
automated analysis support for tensorflow models

### LINKED ISSUE #75: Add automated analysis support for tensorflow models
- [x] I checked to make sure that this issue has not already been filed.

- [x] I'm reporting the issue to the correct repository (for multi-repository projects) **(optional)**

# Expected Behavior:
There should be a class for automated evaluation of tensorflow models that will accept the model's history as an input and make the roc curves, confusion matrix and pr curves. The class can have an additional method to create a doc file with those plots. 

# Solution: 
Similar to like Auto_Results class.

### PR DESCRIPTION
**Pull request by ArkaprabhaChakraborty**
# Description:
A class for automated evaluation of tensorflow models that will accept the model's history as an input and make the roc curves, confusion matrix and pr curves. The class can have an additional method to create a doc file with those plots.

Fixes #75  

## Type of change:

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

 
# Checklist:

- [x] My code follows the style guidelines of this project.
- [x] I have performed a self-review of my own code.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] I have made corresponding changes to the documentation.
- [x] My changes generate no new warnings.
- [x] I have added tests that prove my fix is effective or that my feature works.
- [x] New and existing unit tests pass locally with my changes.
- [x] Any dependent changes have been merged and published in downstream modules.

### CODE REVIEW

File: flow2ml/Tf_Results.py [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+        lw = 2
+        plt.figure(figsize=(50, 20))
+        plt.plot(fpr, tpr, color='darkorange',
+        lw=lw, label='ROC curve (area = %0.2f)' % roc_auc)
+        plt.plot([0, 1], [0, 1], color='navy', lw=lw, linestyle='--')
+        plt.xlabel('False Positive Rate')
+        plt.ylabel('True Positive Rate')
+        plt.title('Receiver operating characteristic')
+        plt.legend(loc="lower right")
+        #plot_roc_curve(self.model, self.x_test, self.y_test)
```
Comment (chebroluharika) [inline flow2ml/Tf_Results.py:48]: Remove these commented lines

File: flow2ml/Tf_Results.py [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+import tensorflow as tf
+import seaborn as sn
+
+class Tf_Results:
+    def __init__(self,model,validtion_generator):
+        '''
+        Initialises the class variables.
+        '''
+        self.model = model          #stores the model
+        self.validation_generator = validtion_generator
```
Comment (chebroluharika) [inline flow2ml/Tf_Results.py:21]: What is validation_generator here. Please describe the same in params section in documentation part

File: flow2ml/Tf_Results.py [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+        plt.xlabel('Recall')
+        plt.ylabel('Precision')
+        plt.title('Precision Recall Curve')
+        
+        # plot_precision_recall_curve(self.model,self.x_test,self.y_test)
+        # plt.title('Precision Recall Curve', size = 35)
+        plt.savefig(self.results_path+'/'+figure_name)
+
+    def tf_get_results_docx(self,file_name="report.docx"):
+        '''
```
Comment (chebroluharika) [inline flow2ml/Tf_Results.py:87]: What if already there is a file with name “report.docx”

File: flow2ml/Tf_Results.py [hunk 4]
```diff
@@ -0,0 +1,129 @@
...
+        pr, tpr, _ = precision_recall_curve(labels, preds)
+        lw = 2
+        plt.figure(figsize=(50,20))
+        plt.plot(tpr, pr, color='darkorange',
+        lw=lw)
+        plt.xlabel('Recall')
+        plt.ylabel('Precision')
+        plt.title('Precision Recall Curve')
+        
+        # plot_precision_recall_curve(self.model,self.x_test,self.y_test)
```
Comment (chebroluharika) [inline flow2ml/Tf_Results.py:82]: Remove these commented lines

**Review by chebroluharika — COMMENTED** [review]
Please attach sample docx that gets generated using this code

File: flow2ml/Tf_Results.py (hunk 1 above)
Comment (ArkaprabhaChakraborty) [reply flow2ml/Tf_Results.py:48]: okay I'll do that

File: flow2ml/Tf_Results.py (hunk 3 above)
Comment (ArkaprabhaChakraborty) [reply flow2ml/Tf_Results.py:87]: the filename can be changed by the user.

File: flow2ml.egg-info/SOURCES.txt [hunk 5]
```diff
@@ -1,10 +1,12 @@
 README.md
 setup.py
+flow2ml/Auto_Results.py
 flow2ml/Data_Augumentation.py
 flow2ml/Data_Loader.py
 flow2ml/Filters.py
 flow2ml/Flow.py
 flow2ml/__init__.py
+flow2ml/test.py
```
[the remark this answers is not in the document]
Comment (ArkaprabhaChakraborty) [reply flow2ml.egg-info/SOURCES.txt]: done

File: flow2ml/Tf_Results.py (hunk 4 above)
Comment (ArkaprabhaChakraborty) [reply flow2ml/Tf_Results.py]: done

File: flow2ml/Tf_Results.py (hunk 2 above)
Comment (ArkaprabhaChakraborty) [reply flow2ml/Tf_Results.py:19]: Validation generator is the validation dataset object generated by tensorflow Image generator. This something that solely belongs to the domain of knowledge of tensorflow users. I don't think we need to document this as it's available in tensorflow.

Comment (chebroluharika) [reply flow2ml/Tf_Results.py:19]: Yes, thats understood
But we can’t assume things just like that.
As a part of documentation for functions, you should mention clearly what each argument meant

File: flow2ml/Tf_Results.py (hunk 3 above)
Comment (chebroluharika) [reply flow2ml/Tf_Results.py:80]: Yes, okay

**Comment by ArkaprabhaChakraborty** [comment]
[report.docx](https://github.com/flow2ml/Flow2ML/files/6495626/report.docx)
Sample report

File: flow2ml/Tf_Results.py (hunk 2 above)
Comment (ArkaprabhaChakraborty) [reply flow2ml/Tf_Results.py]: Alright made it.

**Review by chebroluharika — APPROVED** [review]
LGTM

**Comment by ArkaprabhaChakraborty** [comment]
LGTM?

**Pull request merged**